core/src/main/java/com/linecorp/armeria/client/circuitbreaker/CircuitBreakerStrategy.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.circuitbreaker;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.function.BiFunction;
+
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.HttpStatusClass;
+import com.linecorp.armeria.common.Response;
+
+/**
+ * Determines whether a {@link Response} should be reported as a success or a failure to a
+ * {@link CircuitBreaker}.
+ *
+ * @param <T> the response type
+ */
+@FunctionalInterface
+public interface CircuitBreakerStrategy<T extends Response> {
+
+    /**
+     * Returns the {@link CircuitBreakerStrategy} that determines a {@link Response} as successful
+     * when its {@link HttpStatus} is not {@link HttpStatusClass#SERVER_ERROR} and there was no
+     * {@link Exception} raised.
+     */
+    static CircuitBreakerStrategy<HttpResponse> onServerErrorStatus() {
+        return onStatus(
+                (status, thrown) -> thrown == null &&
+                                    (status == null || status.codeClass() != HttpStatusClass.SERVER_ERROR));
```

Comment:
I'm not sure but it seems that `status != null && status.codeClass() != HttpStatusClass.SERVER_ERROR` more makes sense. WDYT?